File: C#_Databases_Fundamentals/DB Exam 19.11/MassDefect.Models/Person.cs
namespace MassDefect.Models
{
    using System.Collections;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;

    public class Person
    {
        private ICollection<Anomaly> anomalies;

        public Person()
        {
            this.anomalies = new HashSet<Anomaly>();
        }

        [Key]
        public int Id { get; set; }

        [Required]
        public string Name { get; set; }

        [ForeignKey("HomePlanet")]
        public int HomePlanetId { get; set; }

        public virtual Planet HomePlanet { get; set; }
        
        public virtual ICollection<Anomaly> Anomalies
        {
            get { return this.anomalies; }
            set { this.anomalies = value; }
        }
    }
}

File: Java_Fundamentals_January_2017/OOP_Basics/Exersices/01. Defining Classes/src/OpinionPoll/Person.java
package OpinionPoll;

public class Person {
    public String name;
    public int age;

    public Person (){
        this ("No name", 1);
    }

    public Person (int age) {
        this ("No name", age);
    }

    public Person (String name, int age){
        this.name = name;
        this.age = age;
    }
}


